docs/internals/data-structures.rst

```diff
@@ -998,18 +1008,32 @@ key file, wrapped using the standard ``textwrap`` module with a header.
 The header is a single line with a MAGIC string, a space and a hexadecimal
 representation of the repository id.
 
+.. _data-compression:
+
 Compression
 -----------
 
-Borg supports the following compression methods:
+Borg supports the following compression methods, each identified by a type
+byte:
 
-- none (no compression, pass through data 1:1)
-- lz4 (low compression, but super fast)
+- none (no compression, pass through data 1:1), identified by 0x00
+- lz4 (low compression, but super fast), identified by 0x01
 - zstd (level 1-22 offering a wide range: level 1 is lower compression and high
-  speed, level 22 is higher compression and lower speed) - since borg 1.1.4
+  speed, level 22 is higher compression and lower speed) - since borg 1.1.4,
+  identified by 0x03
 - zlib (level 0-9, level 0 is no compression [but still adding zlib overhead],
-  level 1 is low, level 9 is high compression)
-- lzma (level 0-9, level 0 is low, level 9 is high compression).
+  level 1 is low, level 9 is high compression), identified by 0x05
+- lzma (level 0-9, level 0 is low, level 9 is high compression), identified
+  by 0x02.
+
+The type byte is followed by a byte indicating the compression level.
+
+.. note::
+
+    Borg 1.x (and attic) used to write zlib chunks without dedicated type
+    and level bytes. That means that in archives created using these tools,
+    such legacy zlib chunks may exist. These chunks are identified by the
+    lowest nibble of the first byte being 0x08.
```

Comment:
Correct, but we do not need this note in borg2 docs. Stuff from legacy repos will get transferred into new repos using borg2's  `transfer` command, which also will convert ZLIB_legacy to ZLIB on the fly.

Shorter docs are also a feature, thus I prefer not to describe stuff that does not apply any more (I also removed the AES-CTR related docs, AES-CTR is gone, except when borg transfer reads from an old repo).

Hmm, it seems I overlooked these encryption.* files - are they still references from the text?

Reply:
Aha, I was not aware that `borg transfer` removes that. In that case, fair enough, I dropped that just now.

`encryption.png` is used in the `Legacy modes` section. I found that graphic quite useful to understand the layering of Borg objects (encryption -> compression -> data). It would be good to have an updated version for the AEAD mode.

Reply:
I added a new diagram a while ago.

Reply:
Ah, I hadn't seen that because I obviously looked at the stable docs for my experiments :-).

Would you like to get rid of the Legacy modes section altogether anyway? In that case I'll just drop the commit fixing the graphic.

Reply:
Yes, guess we can get rid of the legacy modes section. This also means slight changes elsewhere (like "both modes" and "aead modes").

Reply:
I'll merge it now and clean up afterwards.